lib/travis/build/git/clone.rb

```diff
@@ -17,7 +17,17 @@ def apply
 
           def clone_or_fetch
             sh.if "! -d #{dir}/.git" do
-              sh.cmd "git clone #{clone_args} #{data.source_url} #{dir}", assert: true, retry: true
+              if sparseCheckout?
+                sh.cmd "git init #{dir}", assert: true, retry: true
+                sh.cmd "git -C #{dir} git config core.sparseCheckout true", assert: true, retry: true
+                sh.cmd "echo #{sparseCheckout} >> #{dir}/.git/info/sparseCheckout", assert: true, retry: true
```

Comment:
Should be like this:
https://github.com/cdnjs/cdnjs/blob/master/tools/sparse-checkout.template

A list of patterns to match the files need to be checkout

We have a short note here:
https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md